### PR TITLE
feat(ranges): introduce MultiThumbRange component

### PR DESCRIPTION
**Pull request by austin94**
## Description

This PR adds a new `MultiThumbRange` component. It follows the [Two-Thumb Slider W3C Pattern](https://www.w3.org/TR/wai-aria-practices/#slidertwothumb).

Due to the complexity of the drag functionality and it's tight coupling to how we structure our sliders visually, I've chosen to not break this component into separate Container and View components.

@jzempel @ryanseddon any thoughts on this? If more customization is needed in the future we can re-evaluate the split.

## Detail

screen shot 2019-01-22 at 11 24 18 am

Pre-published at https://garden.zendesk.com/react-components/ranges

## Checklist

* [ ] :ok_hand: design updates are Garden Designer approved (add the
      designer as a reviewer)
* [ ] :nail_care: view component styling is based on a Garden CSS
  component
* [ ] :globe_with_meridians: Styleguidist demo is up-to-date (`yarn start`)
* [ ] :arrow_left: renders as expected with reversed (RTL) direction
* [ ] :guardsman: includes new unit and snapshot tests
* [ ] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
   },
   "rules": {
     "prettier/prettier": "error",
-    "indent": ["error", 2],
+    "indent": ["error", 2, { "SwitchCase": 1 }],
```
Comment (jzempel) [inline .eslintrc.json]: This is the existing `eslint-config` rule: https://github.com/zendeskgarden/eslint-config/blob/master/rules/stylistic-issues.js#L58. Can just be deleted.

Comment (austin94) [reply .eslintrc.json]: I'm not sure what's going on with our custom config, but it seems prettier is fighting the config on this one and can't seem to align on anything. 

Lets meet offline to walk through this.

**Comment by ryanseddon** [comment]
Seeing some weird behaviour if I get silly with min/max values
multi-slider

File: packages/ranges/src/elements/MultiThumbRange.js [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+
+const StyledThumb = styled.div.attrs({
+  'data-test-id': 'thumb',
+  className: props =>
+    classNames(RangeStyles['c-range__slider__track__rail__thumb'], {
+      [RangeStyles['is-focused']]: props.isFocused
+    })
+})``;
+
+class MultiThumbRange extends PureComponent {
```
Comment (ryanseddon) [inline packages/ranges/src/elements/MultiThumbRange.js]: Technically we still support react 0.14.0 and `PureComponent` came in v15.3.0.

File: packages/ranges/src/elements/MultiThumbRange.js [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+  addRangeResizeEvents = () => {
+    window.addEventListener('resize', this.onWindowResize);
+  };
+
+  removeRangeResizeEvents = () => {
+    window.removeEventListener('resize', this.onWindowResize);
+  };
+
+  addDragEvents = () => {
+    document.addEventListener('mousemove', this.onDocumentMouseMove);
```
Comment (ryanseddon) [inline packages/ranges/src/elements/MultiThumbRange.js]: Should we use `getDocument` here?

File: packages/ranges/src/elements/MultiThumbRange.js (hunk 2 above)
Comment (austin94) [reply packages/ranges/src/elements/MultiThumbRange.js]: I've updated this to use the standard `Component`. I was already checking for extra renders to help with `onChange` management so no other changes were needed.

File: packages/ranges/src/elements/MultiThumbRange.js (hunk 3 above)
Comment (austin94) [reply packages/ranges/src/elements/MultiThumbRange.js]: Updated

File: .eslintrc.json (hunk 1 above)
Comment (jzempel) [reply .eslintrc.json]: If there are plugin rules unintentionally overriding our config, then I think the `extends` section needs to be reordered, like so:

```json
{
  "extends": [
    "prettier",
    "@zendeskgarden",
    "@zendeskgarden/eslint-config/plugins/notice.js",
    "plugin:react/recommended",
    "plugin:jest/recommended",
    "plugin:jsx-a11y/recommended"
  ]
}
```

based on https://eslint.org/docs/user-guide/configuring#extending-configuration-files

**Comment by austin94** [comment]
> Seeing some weird behaviour if I get silly with min/max values

@ryanseddon I've added in some bounded values on in our render. I'm not able to get propType validators to consistently flag issues, but the rendered thumbs won't jump out of the page at least.

Since these values would have to be manually set and the `onChange` callback is already bounding the values I think this should be good.

**Pull request merged**